tools/CoqMakefile.in

```diff
@@ -724,9 +724,18 @@ $(addsuffix .d,$(MLPACKFILES)): %.mlpack.d: %.mlpack
 	$(SHOW)'COQDEP $<'
 	$(HIDE)$(COQDEP) $(OCAMLLIBS) -c "$<" $(redir_if_ok)
 
+# If this makefile is created using a _CoqProject we have coqdep get
+# the list of files from it. This avoids argument length limits for
+# pathological projects.
+ifeq (,@PROJECT_FILE@)
+COQDEP_VFILES:=$(VFILES)
+else
+COQDEP_VFILES:=-f @PROJECT_FILE@
+endif
+
 $(VDFILE).d: $(VFILES)
 	$(SHOW)'COQDEP VFILES'
-	$(HIDE)$(COQDEP) $(COQLIBS) -dyndep var -c $(VFILES) $(redir_if_ok)
+	$(HIDE)$(COQDEP) $(COQLIBS) -dyndep var -c $(COQDEP_VFILES) $(redir_if_ok)
```

Comment:
what something along the lined of
```
tmp=`mktemp`; echo $VFILES > $tmp; coqdep -f $tmp; rm $tmp
```
so that VFILES still stays authoritative?

Reply:
That won't work; we'll just get the command line length limit with `echo`.  Instead, we can do ```tmp=`mkstemp`; $(foreach f,$(VFILES),echo '"$$f"' >> "$$tmp"; coqdep -f $$tmp; rm -f "$$tmp"``` (with appropriate stuff to both remove the file and deal with coqdep failing).

Reply:
yep